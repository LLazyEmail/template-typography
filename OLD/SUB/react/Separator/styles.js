export const styles = {
  div: {
    textAlign: 'center'
  },
  span1: {
    fontSize: '16px'
  },
  span2: {
    fontFamily: 'trebuchet ms,lucida grande,lucida sans unicode,lucida sans,tahoma,sans-serif'
  }
}