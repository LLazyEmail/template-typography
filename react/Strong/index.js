import Strong from './Strong.jsx';

export default Strong;
