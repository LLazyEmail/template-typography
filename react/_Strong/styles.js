export const styles = {
  strong: {
    fontWeight: 'bolder'
  }
};