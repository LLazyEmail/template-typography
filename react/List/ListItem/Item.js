import React from 'react';
import styles from './styles';

const ListItem = ({ children }) => (
  <li style={styles.li}>
    <p dir="ltr" role="presentation" style={styles.presentation} data-testid="listItemTest">

      {children}
      
    </p>
  </li>
);

export default ListItem;
