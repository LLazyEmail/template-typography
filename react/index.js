/* eslint import/extensions: 0 */

import Divider from './Divider/Divider.jsx';

import HeadingThree from './HeadingThree/HeadingThree.jsx.js';

import Italic from './Italic/Italic.jsx';
import Link from './Link/Link.jsx';
import List from './List/List.jsx';
import HeadingOne from './HeadingOne/HeadingOne.jsx';
import Paragraph from './Paragraph/Paragraph.jsx';
import Separator from './Separator/Separator.jsx';
import Strong from './Strong/Strong.jsx';
import Subtitle from './Subtitle/Subtitle.jsx';

export {
    Divider, Heading, Italic, 
    Link, List, MainTitle, 
    Paragraph, Separator, 
    Strong, Subtitle
};
