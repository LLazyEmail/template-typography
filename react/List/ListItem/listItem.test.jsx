import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import styles from './styles';
import ListItem from './item';

configure({ adapter: new Adapter() });

describe('ListItem component', () => {
    it('<ListItem> receives styles', () => {
        const wrapper = shallow(<ListItem />);
        const li = wrapper.find('li');
        const p = wrapper.find('p');
        expect(li.props()).toHaveProperty('style', styles.li);
        expect(p.props()).toHaveProperty('style', styles.presentation);
    })

    it('<ListItem > should containe <li> element', () => {
        const wrapper = shallow(<ListItem />);
        expect(wrapper.html()).toContain('li');
    });

    it('checking <p> attributes', () => {
        const wrapper = shallow(<ListItem />);
        const p = wrapper.find('p');
        expect(p.props()).toHaveProperty('dir', 'ltr')
    }) 

    it('checking <p> attributes', () => {
        const wrapper = shallow(<ListItem />);
        const p = wrapper.find('p');
        expect(p.props()).toHaveProperty('role', 'presentation')
    }) 

    it('checking <p> attributes', () => {
        const wrapper = shallow(<ListItem />);
        const p = wrapper.find('p');
        expect(p.props()).toHaveProperty('data-testid', 'listItemTest')
    }) 

    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ListItem />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('renders correctly', () => {
        const tree = create(<ListItem><span>Some text</span></ListItem>);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('compoents props should contain children(h1) elements', () => {
        const component = create(<ListItem><span>Some text</span></ListItem>);
        const { root } = component;
        expect(root.props.children.type).toBe('span');
    });

    test('<p> has correct children elements', () => {
        const { getByTestId } = render(<ListItem><span>Hello</span></ListItem>);
        const p = getByTestId('listItemTest');

        expect(p.children[0].textContent).toBe('Hello');
    });
});
