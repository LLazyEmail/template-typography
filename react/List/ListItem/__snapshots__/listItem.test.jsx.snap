// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders correctly 1`] = `
<li
  style={
    Object {
      "WebkitTextSizeAdjust": "100%",
      "msTextSizeAdjust": "100%",
      "msoLineHeightRule": "exactly",
    }
  }
>
  <p
    data-testid="listItemTest"
    dir="ltr"
    role="presentation"
    style={
      Object {
        "WebkitTextSizeAdjust": "100%",
        "color": "#111111",
        "fontFamily": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        "fontSize": "18px",
        "lineHeight": "125%",
        "margin": "10px 0",
        "msTextSizeAdjust": "100%",
        "msoLineHeightRule": "exactly",
        "padding": 0,
        "textAlign": "left",
      }
    }
  >
    <span>
      Some text
    </span>
  </p>
</li>
`;
